src/transform/src/monotonic.rs

```diff
@@ -116,6 +116,35 @@ impl MonotonicFlag {
                     }
                     result
                 }
+                MirRelationExpr::LetRec { ids, values, body } => {
```

Comment:
It may, but I think it is possibly lots more complicated than this. Datalog has a lot more going for its asynchronous rules than just that it is monotonic. Also, DD is already "monotonic" in the sense that we only ever produce more `(data, time, diff)` conclusions, but this doesn't allow it or us to re-order operators in a loop without great care. There is an asynchronous DD executor (original Naiad was this) but it still has a synchronous semantics (e.g. as if `LetRec` was executed in order repeatedly). So .. may or may not unlock program transformations, is I guess what I mean.

There is a property "confluence" that folks often use, that you arrive at the same result from the same start no matter how you try and apply rules. Reasoning about that might allow us to transform the order of bindings, move things around and extract DAG structure that happens to go around the loop.

TBH, I'm hoping to start that we 1. don't mis-optimize things, and 2. get the bare bones improvements like 
```sql
data (a int, b timestamp, c text) AS (SELECT * FROM best UNION base_case),
best (a int, b timestamp, c text) AS (SELECT distinct on(a) a, b, c FROM foo ORDER BY a, b DESC)
```
being monotonic.